--- src/main/java/app/controllers/MainWebController.java
+++ src/main/java/app/controllers/MainWebController.java
@@ -83,6 +83,7 @@
     auctionScrollPane.setHbarPolicy(ScrollPane.ScrollBarPolicy.AS_NEEDED);
     auctionScrollPane.setVbarPolicy(ScrollPane.ScrollBarPolicy.AS_NEEDED);
     logInLabel.setText("Hi "+ user.getUserName());
+    displayAuctionList(999, "TEST AUCTION", "WAITING FOR NETWORK");
   }
 
   @FXML
